tools/manually-handling-sharding-ddl-locks.md

```diff
@@ -0,0 +1,415 @@
+3. Ask all other DM-workers to skip the DDL and update the checkpoints of corresponding sharded tables after the owner successfully executes the DDL. 
+
+Currently, the above unlocking process is not atomic. Therefore, after the owner successfully executes the DDL, if a DM-worker restarts during the period of asking other DM-workers to skip the DDL, then the DM-worker might fail to skip the DDL. 
+
+At this point, the lock information on `DM-master` has been removed and the restarted DM-worker will continue to synchronize the DDL, but as other DM-workers (including the previous owner) has synchronized the DDL and continued the synchronization process, this DM-worker will never see the DDL lock be unlocked automatically. 
+
+##### Manual solution
+
+Suppose that now we have the same upstream and downstream table structures and the same demand for merging tables and synchronization as in the manual solution of [Some DM-workers go offline](#Some-DM-workers-go-offline).
+
+When `DM-master` automatically executes the unlocking process, the owner (`DM-worker-1`) successfully executes the DDL and continues the synchronization process, and the DDL lock information has been removed from `DM-master`. But at this point, if `DM-worker-2` restarts during the period of asking `DM-worker-2` to skip the DDL, then the skipping process might fail. 
+
+After `DM-worker-2` restarts, it will try to synchronize the waiting DDL lock before it restarted. At this point, a new lock will be created on `DM-master`, and the DM-worker will become the owner of the lock (other DM-workers have executed/skipped the DDL by now and are continuing the synchronization process).
+
+The operation processes are:
+
+1. Use `show-ddl-locks` to confirm if the corresponding lock of the DDL exists on `DM-master`. 
+    - Only the restarted DM-worker (`127.0.0.1:8263`) is at the `synced` state. 
+    ```bash
+    » show-ddl-locks
+    {
+        "result": true,
+        "msg": "",
+        "locks": [
+            {
+                "ID": "test-`shard_db`.`shard_table`",
+                "task": "test",
+                "owner": "127.0.0.1:8263",
+                "DDLs": [
+                    "USE `shard_db`; ALTER TABLE `shard_db`.`shard_table` ADD COLUMN `c2` int(11);"
+                ],
+                "synced": [
+                    "127.0.0.1:8263"
+                ],
+                "unsynced": [
+                    "127.0.0.1:8262"
+                ]
+            }
+        ]
+    }
+    ```
+2. Use `unlock-ddl-lock` to ask `DM-master` to unlock the lock.
+    - Use the parameter `--worker` to limit the operation to only target at the restarted DM-worker (`127.0.0.1:8263`). 
+    - The DM-worker will try to execute the DDL to the downstream again during the locking process (the owner before restarting has executed the DDL to the downstream), so as to make sure that the DDL can be executed multiple times.
```

Comment:
should be `during the unlocking process`, it's the fault of CN doc.

Reply:
Okay.